src/app/layout.tsx

```diff
@@ -9,14 +9,31 @@ import { getServerSession } from "next-auth";
 import { getLocale } from "./functions/server/l10n";
 import { SessionProvider } from "../contextProviders/session";
 import { authOptions } from "./api/auth/[...nextauth]/route";
+import { getL10n } from "./functions/server/l10n";
 
 const inter = Inter({ subsets: ["latin"] });
 
-export const metadata = {
-  title: "Firefox Monitor",
-  description:
-    "Find out if you’ve been part of a data breach with Firefox Monitor. We’ll help you understand what to do next and continuously monitor for any new breaches.",
-};
+export async function generateMetadata() {
+  const l10n = getL10n();
+  return {
+    title: l10n.getString("brand-fx-monitor"),
+    description: l10n.getString("meta-desc-2"),
+    twitter: {
+      card: "summary_large_image",
+      title: l10n.getString("brand-fx-monitor"),
```

Comment:
I think before we were overriding the social titles in some cases. Not sure if it matters, but 🤷 .
https://github.com/mozilla/blurts-server/blob/d4cb77862d91a61e0debf1032301428c2019ee04/src/views/guestLayout.js#L13-L16

Reply:
Thanks, I am reading about how to override these per-page and merge: https://nextjs.org/docs/app/building-your-application/optimizing/metadata

Reply:
> Thanks, I am reading about how to override these per-page and merge: https://nextjs.org/docs/app/building-your-application/optimizing/metadata

OK so per this doc, the problem is that you can override single keys, but it will not attempt to merge e.g. `openGraph` or `twitter`, so if this is set at the top-level:

```
  return {
    title: l10n.getString("brand-fx-monitor"),
    description: l10n.getString("meta-desc-2"),
    twitter: {
      card: "summary_large_image",
      title: l10n.getString("brand-fx-monitor"),
      description: l10n.getString("meta-desc-2"),
      images: ["/images/og-image.webp"],
    },
    openGraph: {
      title: l10n.getString("brand-fx-monitor"),
      description: l10n.getString("meta-desc-2"),
      siteName: l10n.getString("brand-fx-monitor"),
      type: "website",
      url: process.env.SERVER_URL,
      images: ["/images/og-image.webp"],
    },
  };
```

Then when a page wants to just override a small part of this:

```
  return {
    title: l10n.getString("..."),
    twitter: {
      title: l10n.getString("..."),
      description: l10n.getString("..."),
    },
    openGraph: {
      title: l10n.getString("..."),
      description: l10n.getString("..."),
    },
  };
```

This works fine for `title` (it leaves `description` defined), but the entire `twitter` and `openGraph` objects get overwritten.

There is a suggested way to deal with it, which is a bit trickier for us since we need to localize... for the moment I just duplicated the info on overriding since we don't have very many unique pages, but I'd like to figure out a cleaner solution.

This deep merge approach makes sense to me, we should see if it works for us: https://github.com/vercel/next.js/issues/46899#issuecomment-1472048414

Reply:
https://nextjs.org/docs/app/api-reference/functions/generate-metadata also has some good info on `default`s, `template`s and using `parent` metadata stuff:

```ts
import { Metadata, ResolvingMetadata } from 'next';
 
type Props = { ... };

export async function generateMetadata(
  { params, searchParams }: Props,
  parent: ResolvingMetadata,
): Promise<Metadata> {
  ...
}
```

The previously mentioned https://nextjs.org/docs/app/building-your-application/optimizing/metadata#merging also has a good section on merged metadata and overwriting `openGraph`, `twitter`, and `robots`:

> ### Merging
> Following the [evaluation order](https://nextjs.org/docs/app/building-your-application/optimizing/metadata#ordering), Metadata objects exported from multiple segments in the same route are shallowly merged together to form the final metadata output of a route. Duplicate keys are replaced based on their ordering.
>
> <i>**This means metadata with nested fields such as [`openGraph`](https://nextjs.org/docs/app/api-reference/functions/generate-metadata#opengraph) and [`robots`](https://nextjs.org/docs/app/api-reference/functions/generate-metadata#robots) that are defined in an earlier segment are overwritten by the last segment to define them.**
> &hellip;
> **If you'd like to share some nested fields between segments while overwriting others, you can pull them out into a separate variable: &hellip;**</i>

Unrelated, but interesting read, but here's one of the Next.js source files for metadata stuff: https://github.com/vercel/next.js/blob/canary/packages/next/src/lib/metadata/types/metadata-interface.ts

Reply:
I see no problem with copy-pasting. One other workaround could be to export an object with the defaults and destructure that elsewhere, but especially with the low number of pages we have, I'd lean towards the indirection being more harmful than the duplication.